pkg/cloudprovider/drift.go

```diff
@@ -119,6 +126,16 @@ func (c *CloudProvider) areSecurityGroupsDrifted(ec2Instance *instance.Instance,
 	return "", nil
 }
 
+// Checks if capacity reservations are drifted, by comparing the capacity reservations persisted to the NodeClass to
+// the instance's capacity reservation.
+func (c *CloudProvider) isCapacityReservationDrifted(instance *instance.Instance, nodeClass *v1.EC2NodeClass) cloudprovider.DriftReason {
```

Comment:
Super small nit for commenting/documentation: Can we note down why we made the decision to check this dynamically (specifically the interaction that it has with changing the capacity type of an instance when it's converted from ODCR to OD)